MonoGame.Framework/TitleContainer.cs

```diff
@@ -13,15 +17,118 @@ namespace Microsoft.Xna.Framework
     /// </summary>
     public static partial class TitleContainer
     {
+        private const string CACHE_PATH = "MonoGameCache";
+        private const string MODIFIED_TIMES_FILENAME = "ModifiedTimes.yaml";
+
         static partial void PlatformInit();
 
-        static TitleContainer() 
+        static TitleContainer()
         {
+            Client = new();
             Location = string.Empty;
+            CheckContentServer = false;
+            ContentServerAddress = "http://localhost:7771/";
             PlatformInit();
+            ModifiedTimes = LoadModifiedTimes();
+        }
+
+        /// <summary>
+        /// Determines if the assets will first be attempted to be loaded from a content server.
+        /// </summary>
+        /// <value><c>false</c> by default.</value>
+        public static bool CheckContentServer { get; set; }
+
+        /// <summary>
+        /// The location of the content server for loading assets. To use it enable <see cref="CheckContentServer"/>.
+        /// </summary>
+        /// <value><c>http://localhost:7771/</c> by default.</value>
+        public static string ContentServerAddress { get; set; }
+
+        internal static string Location { get; private set; }
+
+        private static Dictionary<string, long> ModifiedTimes { get; set; }
+
+        private static HttpClient Client { get; set; }

```

Comment:
I'm wondering about the impact of referencing ```HttpClient```, it makes self-contained apps substantially bigger and untrimmable.